lib/spack/spack/build_systems/autotools.py

```diff
@@ -237,6 +238,14 @@ def delete_configure_to_force_update(self):
         if self.force_autoreconf:
             force_remove(self.configure_abs_path)
 
+    @run_before('autoreconf')
+    def add_aclocal_path(self):
+        env = EnvironmentModifications()
+        for dep in self.spec.dependencies(deptype='build'):
+            if os.path.exists(dep.prefix.share.aclocal):
+                env.append_path('ACLOCAL_PATH', dep.prefix.share.aclocal)
+        env.apply_modifications()
```

Comment:
Shouldn't this go in `setup_build_environment`?

Reply:
I  want to avoid replacing `setup_build_environment` with the package,py

Reply:
@tkameyama Ugh, forget my comment above. I saw the title of the PR starting with `gettext` and didn't notice this change was not referring to that package but to the `autotools` build system.

Reply:
After thinking a bit about that, I believe I can employ metaclasses to have this bit of code in `setup_*_environment` without changing any `package.py` implementation.

Reply:
For the record, the approach I propose above can be done by intercepting class creation and patching `setup_*_environment` to add the autotools specific modifications. Said otherwise, with meta-classes we can decorate any method on the fly and programmatically. Not sure it's worth it though, since it's more code than this and more cryptic.